src/sentry/integrations/slack/action_endpoint.py

```diff
@@ -261,7 +285,18 @@ def post(self, request):
                     self.open_resolve_dialog(data, group, integration)
                     defer_attachment_update = True
         except client.ApiError as e:
-            return self.api_error(e, action_type, logging_data)
+
+            unlinking_url = build_unlinking_url(
+                integration.id, group.organization.id, user_id, channel_id, response_url
+            )
+
+            text = UNLINK_IDENTITY_MESSAGE.format(
```

Comment:
@NisanthanNanthakumar I think it would make more sense to have all the logic for determining what error message to show here. So something like:
```
if error.status_code == 403:
   text = ...
else:
   text = ...
```
I think it's a bit confusing to have this logic both in the caller function and inside of `api_error`.